gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -382,12 +364,29 @@ public synchronized void setActive(boolean active) {
               ServiceMetricNames.FAILED_FLOW_METER));
         }
 
+        UserQuotaManager quotaManager = new UserQuotaManager(config);
+        // Before initializing the DagManagerThreads check which dags are currently running before shutdown
+        for (Dag<JobExecutionPlan> dag: dagStateStore.getDags()) {
+          for (DagNode<JobExecutionPlan> dagNode: dag.getNodes()) {
+            if (DagManagerUtils.getExecutionStatus(dagNode) == RUNNING) {
+              // Add all the currently running Dags to the quota limit per user
+              try {
+                quotaManager.checkQuota(dagNode);
+              } catch (IOException e) {
+                // Quota is somehow exceeded with currently running jobs, we should never hit this state normally
+                // but we should avoid stalling the entire service
+                log.error(String.format("Quota exceeded during initialization in DagManager for job name: %s",
```

Comment:
Ah I didn't think about that edge case, nice catch. Even then, it would be during startup so I don't see a reason to fail a job that's still running on Azkaban even if the quota is decreased, at most we should be logging it as a warning and reject additional jobs until the currently running jobs is under the quota.

Reply:
Will add an additional test case for this